### PR TITLE
Spearbit 171

### PR DESCRIPTION
**Pull request by LayneHaber**
## Description

Fixes https://github.com/spearbit-audits/connext-nxtp/issues/171

- Adds `DelayBlocksUpdated`
- Adds `AavePortalFeeUpdated`
- Adds `AavePoolUpdated`
- Adds `MessageSent` emission to `processFromRoot` flow

## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

**Review by liu-zhipeng — APPROVED** [review]

File: packages/deployments/contracts/contracts/messaging/connectors/polygon/PolygonHubConnector.sol [hunk 1]
```diff
@@ -47,10 +47,8 @@ contract PolygonHubConnector is HubConnector, FxBaseRootTunnel {
     emit MessageProcessed(message, msg.sender);
   }
 
-  function _processMessage(bytes memory _data) internal override {
-    // Does nothing, all messages should go through the `_processMessageFromChild` path
-    revert Connector__processMessage_notUsed();
-  }
+  // DO NOT override _processMessage, should revert from `Connector` class. All messages must use the
+  // `processMessageFromRoot` flow.
```
Comment (Saw-mon-and-Natalie) [inline packages/deployments/contracts/contracts/messaging/connectors/polygon/PolygonHubConnector.sol:51]: `processMessageFromRoot` needs to be `_processMessageFromChild` in the comment.

File: packages/deployments/contracts/contracts/messaging/connectors/zksync/ZkSyncHubConnector.sol [hunk 2]
```diff
@@ -129,6 +123,7 @@ contract ZkSyncHubConnector is HubConnector, GasCap {
       processed[_root] = true;
       // update the root on the root manager
       IRootManager(ROOT_MANAGER).aggregate(MIRROR_DOMAIN, _root);
+      emit MessageSent(_message, bytes(""), msg.sender);
```
Comment (Saw-mon-and-Natalie) [inline packages/deployments/contracts/contracts/messaging/connectors/zksync/ZkSyncHubConnector.sol:126]: We need to emit emit `MessageProcessed(_message, msg.sender)`

File: packages/deployments/contracts/contracts/messaging/connectors/Connector.sol [hunk 3]
```diff
@@ -169,7 +169,11 @@ abstract contract Connector is ProposedOwnable, IConnector {
    * @notice This function is used by the AMBs to handle incoming messages. Should store the latest
    * root generated on the l2 domain.
    */
-  function _processMessage(bytes memory _data) internal virtual;
+  function _processMessage(bytes memory _data) internal virtual {
```
Comment (Saw-mon-and-Natalie) [inline packages/deployments/contracts/contracts/messaging/connectors/Connector.sol:172]: unused `_data` can be commented out:

```suggestion
  function _processMessage(bytes memory /* _data */) internal virtual {
```

File: packages/deployments/contracts/contracts/messaging/connectors/polygon/PolygonHubConnector.sol (hunk 1 above)
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/messaging/connectors/polygon/PolygonHubConnector.sol]: fixed! https://github.com/connext/nxtp/pull/2493/commits/0dbd3f98b51a282474c8021d666cb9f70be7afb6

File: packages/deployments/contracts/contracts/messaging/connectors/zksync/ZkSyncHubConnector.sol (hunk 2 above)
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/messaging/connectors/zksync/ZkSyncHubConnector.sol]: fixed! https://github.com/connext/nxtp/pull/2493/commits/13f7262040d36c71f1b84ee5565a4d4207889f23

**Comment by LayneHaber** [comment]
tests failing due to coverage

**Pull request merged**